src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -894,7 +931,11 @@ public static unsafe int IndexOfAny(ref char searchStart, char value0, char valu
                             goto NotFound;
                         }
 
-                        goto IntrinsicsMatch;
```

Comment:
Avoiding goto showed performance improvement on x64. @adamsitnik , feel free to confirm this on your end. If it is useful, we can apply it to other places.